.github/workflows/mainPipeline.yml

```diff
@@ -0,0 +1,164 @@
+name: Main pipeline
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
+    inputs:
+      override:
+        description: 'Version Number'     
+        required: true
+        default: '0.0.0'
+
+jobs:
+  draftRelease:
+    name: Draft release
+    runs-on: ubuntu-latest
+    steps:
+      # Drafts your next Release notes as Pull Requests are merged into "master"
+      - uses: release-drafter/release-drafter@v5
+        if: github.event_name == 'push'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  buildTest:
+    name: Build and test
+    runs-on: windows-latest 
+    needs: draftRelease
+    outputs:
+      buildNumber: ${{ steps.ver.outputs.BUILD_NUMBER }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+
+      - name: GetVersion
+        id: ver
+        shell: pwsh
+        run: |
+              if ("${{github.event.inputs.override}}")
+              {
+                Write-Host "Using manually entered override version"
+                $version = "${{ github.event.inputs.override}}"
+              }
+              else
+              {
+                $header = @{Authorization = 'Bearer ${{ secrets.GITHUB_TOKEN }}' }
+                $endPoint = "https://api.github.com/repos/${{github.repository}}/releases"
+                $info = Invoke-RestMethod -Uri $endPoint -Headers $header
```

Comment:
It would probably be good to make sure the step will fail if this script fails, rather than continuing with a corrupted version number like I suspect it will right now

Reply:
Ah, nevermind this comment; [it appears](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsrun) GitHub Actions prepends `$ErrorActionPreference = 'Stop'` automatically.